### PR TITLE
feature: votebot integration

### PR DESCRIPTION
**Pull request by lucasmenendez**
Integration of https://github.com/vocdoni/votebot

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -350,6 +361,103 @@ func main() {
 	if err := uAPI.Endpoint.RegisterMethod(fmt.Sprintf("%s/{id}.png", imageHandlerPath), http.MethodGet, "public", handler.imagesHandler); err != nil {
 		log.Fatal(err)
 	}
+	// if a bot FID is provided, start the bot background process
+	if botFid > 0 {
```
Comment (p4u) [inline main.go:365]: Please move this to the bot package. The only thing that would go here is the registration of the HTTP handler (which would point to bot.WebHookHandler or something like this).

File: main.go [hunk 2]
```diff
@@ -52,8 +54,11 @@ func main() {
 	flag.Int("pprofPort", 0, "The port to use for the pprof http endpoints")
 	flag.String("web3", "https://mainnet.optimism.io", "Web3 RPC Optimism endpoint")
 
-	//flag.String("neynarSignerUUID", "", "neynar signer UUID")
+	// bot flags
+	flag.Uint64("botFid", 0, "FID to be used for the bot")
 	flag.String("neynarAPIKey", "", "neynar API key")
+	flag.String("neynarSignerUUID", "", "neynar signer UUID")
+	flag.String("neynarWebhookSecret", "cUIJfBD1Jn9oQC3hy_jRL9Dox", "neynar Webhook shared secret")
```
Comment (p4u) [inline main.go:61]: Can this webhook secrtet be public information?

File: bot/bot.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	"time"
+
+	"github.com/vocdoni/vote-frame/farcasterapi"
+	"go.vocdoni.io/dvote/log"
+)
+
+// defaultCoolDown is the default time to wait between casts
+const defaultCoolDown = time.Second * 10
+
+type BotConfig struct {
```
Comment (p4u) [inline bot/bot.go:15]: Document all exported methods, structs and errors please.

File: bot/bot.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		for {
+			select {
+			case <-b.ctx.Done():
+				return
+			default:
+				log.Debugw("checking for new casts", "last-cast", b.lastCast)
+				// retrieve new messages from the last cast
+				messages, lastCast, err := b.api.LastMentions(b.ctx, b.lastCast)
+				if err != nil && err != ErrNoNewCasts {
+					log.Errorf("error retrieving new casts: %s", err)
```
Comment (p4u) [inline bot/bot.go:58]: continue missing?

File: election.go [hunk 5]
```diff
@@ -179,6 +179,46 @@ func (v *vocdoniHandler) votersForElection(msg *apirest.APIdata, ctx *httprouter
 	return ctx.Send(data, http.StatusOK)
 }
 
+func (v *vocdoniHandler) createAndWaitForElection(description *ElectionDescription, census *CensusInfo, profile *FarcasterProfile) (types.HexBytes, error) {
```
Comment (p4u) [inline election.go:182]: This is duplicated code, cannot we reuse the existing ones?

**Review by p4u — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (lucasmenendez) [reply main.go:61]: ups nop :P but it is already revoked

File: election.go (hunk 5 above)
Comment (lucasmenendez) [reply election.go:182]: Is not duplicated, the `createElection` launches the election creation but not waits for the process ends. It is also designed as http handler, not as a helper. I propose that the original handler calls to the helper that i will create

**Comment by lucasmenendez** [comment]
FYI, Neynar webhooks are currently not working.

File: main.go [hunk 6]
```diff
@@ -52,7 +53,10 @@ func main() {
 	flag.Int("pprofPort", 0, "The port to use for the pprof http endpoints")
 	flag.String("web3", "https://mainnet.optimism.io", "Web3 RPC Optimism endpoint")
 
-	//flag.String("neynarSignerUUID", "", "neynar signer UUID")
+	// bot flags
+	flag.Uint64("botFid", 0, "FID to be used for the bot")
+	flag.String("botPrivKey", "", "The bot private key to use for signing the vote (hex)")
+	flag.String("hubEndpoint", "", "The hub endpoint to use")
```
Comment (p4u) [inline main.go:59]: As far as I understand, the current implementation only runs the bot if there is a hubEndpoint defined. That means all neynar webhooks logic is currently useless, isn't it? 

That's ok if its a temporary measure, but you must add a big notice in the code explaining why. Also you should explicitly say on the flags that for the bot to run you need to define a hubEndpoint (also maybe renaming it to botHubEndpoint)

File: poll/poll.go [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+	"strings"
+	"time"
+)
+
+const (
+	optionPrefix    = "-"
+	lineBreakSuffix = "\n"
+)
+
+var DefaultConfig = PollConfig{
```
Comment (p4u) [inline poll/poll.go:16]: Document ALL exported types and methods please. But also make sure everything you export needs to be exported

File: poll/poll.go [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+// !poll
+// <question>
+// - <option 1>
+// - <option 2>
+// - <option 3*>
+// - <option 4*>
+// <duration*>
+// The duration is optional and by default is 24 hours. If the message does not
+// follow the format, an error is returned.
+func ParseString(message string, config PollConfig) (*Poll, error) {
```
Comment (p4u) [inline poll/poll.go:50]: config might be a pointer

File: poll/poll.go [hunk 9]
```diff
@@ -0,0 +1,140 @@
+package poll
```
Comment (p4u) [inline poll/poll.go:1]: I'm not sure what this "poll" package is aimed to do, the name seems confusing to me, please can you add a package description?
If that's something specific for the bot, this should go inside the bot package. If its more generic, we need to modify all the code to use this new poll tipe.

**Review by p4u — CHANGES_REQUESTED** [review]

File: main.go [hunk 10]
```diff
@@ -113,6 +128,12 @@ func main() {
 		"mongoDB", mongoDB,
 		"pollSize", pollSize,
 		"pprofPort", pprofPort,
+		"botFid", botFid,
+		"botPrivKey", botPrivKey,
```
Comment (p4u) [inline main.go:132]: No secret inputs must be send to LOG

**Review by p4u — APPROVED** [review]

**Pull request merged**